cmd/collectors/restperf/plugins/volumetopclients/volumetopclients.go

```diff
@@ -0,0 +1,409 @@
+package volumetopclients
+
+import (
+	"github.com/netapp/harvest/v2/cmd/collectors"
+	"github.com/netapp/harvest/v2/cmd/poller/plugin"
+	"github.com/netapp/harvest/v2/cmd/tools/rest"
+	"github.com/netapp/harvest/v2/pkg/conf"
+	"github.com/netapp/harvest/v2/pkg/matrix"
+	"github.com/netapp/harvest/v2/pkg/set"
+	"github.com/netapp/harvest/v2/pkg/util"
+	"github.com/tidwall/gjson"
+	"math"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type VolumeInterface interface {
+	fetchTopClients(volumes []string, svms []string, metric string) ([]gjson.Result, error)
+	fetchVolumesWithActivityTrackingEnabled() (*set.Set, error)
+	processTopClients(data *matrix.Matrix) error
+}
+
+const (
+	keyToken                 = "?#"
+	topClientReadOPSMatrix   = "volume_top_clients_read_ops"
+	topClientWriteOPSMatrix  = "volume_top_clients_write_ops"
+	topClientReadDataMatrix  = "volume_top_clients_read_data"
+	topClientWriteDataMatrix = "volume_top_clients_write_data"
+	defaultTopN              = 5
+	maxTopN                  = 50
+)
+
+var opMetrics = []string{
+	"ops",
+}
+var dataMetrics = []string{
+	"data",
+}
+
+type Volume struct {
```

Comment:
Would a better name be VolumeTopClients or TopClients? Volume is a bit confusing since there are n volumes per cluster. This struct is really the plugin, which there will be one per poller instance. Volume makes it sound like there will be n per poller instance. On the other hand, I guess we do the same in volume.go for the Rest/Zapi volume plugin.